Slam 55 - Adding stereo cam

Federico-PizarroBejarano opened the pull request:
Added stereo cam, disparity, point cloud, and depth.

Federico-PizarroBejarano committed:
Added stereo_image_proc to process images from cameras

Federico-PizarroBejarano committed:
Fixed orientation of point-cloud, now the focal distance needs to be adjusted I think

Federico-PizarroBejarano committed:
Moving namespace to zed

Federico-PizarroBejarano committed:
Fixed point cloud

Federico-PizarroBejarano committed:
Adding depth

Federico-PizarroBejarano committed:
Fixing spaces

Federico-PizarroBejarano committed:
:Merge branch 'dev' into SLAM-38-AddingStereoCam

Federico-PizarroBejarano committed:
Replacing tabs with 4 spaces

Federico-PizarroBejarano committed:
Turning off legacy mode

erickmu1 commented on slam-ws/src/motor_control/launch/motor_control.launch:
```diff
@@ -1,7 +1,7 @@
 
   <!-- Ports -->
-	<arg name="motor_port" default="/dev/ttyACM0"/>
+    <arg name="motor_port" default="/dev/ttyACM0"/>
   <arg name="encoder1_port" default="/dev/ttyACM1"/>
   <arg name="encoder2_port" default="/dev/ttyACM2"/>
```
why did the indent change?

Federico-PizarroBejarano replied:
I switched all the tabs in all our files to 4 spaces

erickmu1 commented on slam-ws/src/igvc_world/models/igvc_basic_ground/materials/scripts/igvc_basic_ground.material:
```diff
@@ -1,7 +1,7 @@
 {
-	technique
-	{
+    technique
```
there are weird indent changes in this file

erickmu1 commented on slam-ws/src/caffeine/urdf/inertias.xacro:
```diff
@@ -1,25 +1,25 @@
+            ixz="0.0" 
                         iyy="${mass * (width*width + length*length)/12.0}" 
                         iyz="0.0" 
-			izz="${mass * (width*width + height*height)/12.0}"
-		/>
-				
-	</xacro:macro>
+            izz="${mass * (width*width + height*height)/12.0}"
+        />
+                
+    </xacro:macro>
 
-	<xacro:macro name="cylinder_inertia" params="mass length radius">
-		<inertia
-			ixx="${mass * (3*radius*radius + length*length)/12.0}" 
-			ixy="0.0" 
-			ixz="0.0" 
+    <xacro:macro name="cylinder_inertia" params="mass length radius">
+        <inertia
+            ixx="${mass * (3*radius*radius + length*length)/12.0}" 
+            ixy="0.0" 
+            ixz="0.0" 
                         iyy="${mass * (3*radius*radius + length*length)/12.0}" 
-			iyz="0.0" 
-			izz="${0.5*mass*radius*radius}"
-		/>
-	</xacro:macro>
```
there are weird indent changes in this file

erickmu1 commented on slam-ws/src/caffeine/launch/caffeine_setup.launch:
```diff
@@ -28,7 +28,7 @@
         <param name="serial_port" value="$(arg lidar_port)"/>
     </node>
 
-	<!-- Add Lidar Filtering -->
```
there are weird indent changes in this file

erickmu1 commented on slam-ws/src/motor_control/src/cmd_vel_to_motor.cpp:
```diff
@@ -53,7 +53,7 @@ int main(int argc, char **argv)
     ros::Rate loop_rate(100);
 
     std_msgs::Float64 rvelmsg;
-    std_msgs::Float64 lvelmsg;	
```
there are weird space changes in this file

erickmu1 commented on slam-ws/src/rosserial_python/src/rosserial_python/SerialClient.py:
```diff
@@ -241,7 +241,7 @@ def listen(self):
             self.socket = clientsocket
             self.isConnected = True
 
-            if (self.fork_server == True):	# if configured to launch server in a separate process
```
there are weird space changes in this file

erickmu1 commented:
Can you give the topics the correct name according to the namespace `/zed/zed_node/`. See [here](https://www.stereolabs.com/docs/ros/) for the names of the topics you published. Also idk if you need rectified images from the stereo cam... You can also move this into the simulate ZED stereo cam issue

Federico-PizarroBejarano commented:
> Can you give the topics the correct name according to the namespace `/zed/zed_node/`. See [here](https://www.stereolabs.com/docs/ros/) for the names of the topics you published. Also idk if you need rectified images from the stereo cam... You can also move this into the simulate ZED stereo cam issue

So the new namespace needs to be /zed/zed_node? where does it say that? The rectified images are produced automatically.

Federico-PizarroBejarano committed:
Added visual odometry

Federico-PizarroBejarano commented:
@erickmu1 I added the visual odom package and added it to the gazebo launch file. the topics are created and seem to be working but I don't know how to test it

erickmu1 commented:
> > Can you give the topics the correct name according to the namespace `/zed/zed_node/`. See [here](https://www.stereolabs.com/docs/ros/) for the names of the topics you published. Also idk if you need rectified images from the stereo cam... You can also move this into the simulate ZED stereo cam issue
> 
> So the new namespace needs to be /zed/zed_node? where does it say that? The rectified images are produced automatically.

this is actually happens when you run the launch files they provide, all the topics are namespaced by `/zed` and it seems there is another layer of namespacing under `/zed_node`. Hence when you launch the stereo cam, you see topics such as `/odom` being published under `/zed/zed_node/odom`. Really this is just how it is when we launch stuff and would make simulation mimic reality a little more (plus less worrying over changing topic names).

erickmu1 commented:
> @erickmu1 I added the visual odom package and added it to the gazebo launch file. the topics are created and seem to be working but I don't know how to test it

You can move the robot and see if the odom readings match the direction you are moving in. Furthermore, you could compare the values between visual_odom and `odom/filtered` and see if they are about the same/consistent when the robot is moving. To move the robot u could either give a nav goal or use teleop keyboard, I'll leave it up to you

Federico-PizarroBejarano committed:
Using namespace /zed/zed_node

Federico-PizarroBejarano committed:
Fixing cameras and using correct resolution
